mcs/class/System.Private.CoreLib/System.Collections.Generic/EqualityComparer.cs

```diff
@@ -1,17 +1,58 @@
+using System.Runtime.CompilerServices;
+
 namespace System.Collections.Generic
 {
 	partial class EqualityComparer<T>
 	{
-		public static EqualityComparer<T> Default {
-			get {
-				throw new NotImplementedException ();
-			}
+        static volatile EqualityComparer<T> defaultComparer;
```

Comment:
Nit: The indentation is off in the whole file. Makes it hard to read on GitHub web.